src/app/server/CommissioningWindowManager.cpp

```diff
@@ -305,7 +308,7 @@ CHIP_ERROR CommissioningWindowManager::OpenEnhancedCommissioningWindow(Seconds16
     CHIP_ERROR err = OpenCommissioningWindow(commissioningTimeout);
     if (err != CHIP_NO_ERROR)
     {
-        Cleanup();
+        Cleanup(/* expireFailSafeTimer */ true);
```

Comment:
Likewise, I don't think a fail-safe can be armed here.

Reply:
All the times I set it to true here I just figured it would be safe to disable since we only do it if fail-safe is enabled. But you've made a very compelling argument that these should all be false so we can entirely remove if from the Cleanup(). I also made Cleanup() private now since we have only used it internally